pkg/component/observability/logging/vali/vali.go

```diff
@@ -805,23 +817,28 @@ func (v *vali) getKubeRBACProxyClusterRoleBinding(serviceAccountName string) *rb
 
 func (v *vali) getValitailClusterRole() *rbacv1.ClusterRole {
```

Comment:
As we replace `valitail` with `OpenTelemetry Collector` based on a _feature gate_ ( i.e. both options should be handled until the gate graduates to GA ), I was wondering if the `func` should be renamed with a more generic:
  
```suggestion    
func (v *vali) getLoggingAgentClusterRole() *rbacv1.ClusterRole {
```

or alike, to indicate that both

- `gardener.cloud:logging:valitail`
- `gardener.cloud:logging:opentelemetry-collector`

cases are handled within it, similar to what 

```go
func (v *vali) getIngress(secretName string) *networkingv1.Ingress {
```

does. WDYT ?

Reply:
Absolutely, good catch!